app/controllers/recipe_foods_controller.rb

```diff
@@ -0,0 +1,69 @@
+class RecipeFoodsController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  # GET /recipe_foods or /recipe_foods.json
+  def index
+    @recipe_foods = RecipeFood.all
+  end
+
+  # GET /recipe_foods/1 or /recipe_foods/1.json
+  def show; end
+
+  # GET /recipe_foods/new
+  def new
+    @recipe_food = RecipeFood.new
+  end
+
+  # GET /recipe_foods/1/edit
+  def edit; end
+
+  # POST /recipe_foods or /recipe_foods.json
+  def create
+    @recipe = Recipe.find(params[:recipe_id])
```

Comment:
- [ ] Also here, the N+1 can be solved with a similar code snippet like this ` @recipe = Recipe.includes(:recipe_foods, :user).find(params[:id])`